prow/plugins/lifecycle/lifecycle_test.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lifecycle
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+)
+
+type fakeClient struct {
+	commented bool
+	added     []string
+	removed   []string
+}
+
+func (c *fakeClient) CreateComment(owner, repo string, number int, comment string) error {
+	c.commented = true
+	return nil
+}
+
+func (c *fakeClient) AddLabel(owner, repo string, number int, label string) error {
+	c.added = append(c.added, label)
+	return nil
+}
+
+func (c *fakeClient) RemoveLabel(owner, repo string, number int, label string) error {
+	c.removed = append(c.removed, label)
+	return nil
+}
+
+func TestDeprecatedClose(t *testing.T) {
+	fc := &fakeClient{}
+	gce := &github.GenericCommentEvent{}
+	ticker := make(chan time.Time, 1)
+	deprecatedTick = ticker
+	err := deprecate(fc, "fake", "org", "repo", 1, gce)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if fc.commented {
+		t.Fatalf("should not comment on empty ticker")
+	}
+	ticker <- time.Now()
+	err = deprecate(fc, "fake", "org", "repo", 1, gce)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if !fc.commented {
+		t.Fatalf("must comment on filled timer")
+	}
+
+}
+
+func TestAddLifecycleLabels(t *testing.T) {
+	var testcases = []struct {
+		body    string
+		added   []string
+		removed []string
+		comment bool
+	}{
+		{
+			body:  "/frozen",
+			added: []string{"frozen"},
+		},
+		{
+			body:  "/freeze",
+			added: []string{"frozen"},
+		},
+		{
+			body:  "/stale",
+			added: []string{"stale"},
+		},
+		{
+			body:  "/putrid",
+			added: []string{"putrid"},
+		},
+		{
+			body:  "/rotten",
+			added: []string{"rotten"},
+		},
+		{
+			body:    "/remove-freeze",
+			removed: []string{"frozen"},
+		},
+		{
+			body:    "/stale cancel",
+			removed: []string{"stale"},
+		},
+		{
+			body:    "/remove-stale cancel",
+			comment: true,
+		},
+		{
+			body:    "/remove-stale\n/remove-putrid\n/freeze",
+			added:   []string{"frozen"},
+			removed: []string{"stale", "putrid"},
+		},
```

Comment:
maybe also add a testcase for invalid labels, ensure those are not going to be added/removed

Reply:
Done